### PR TITLE
GSdx-gui: Add debug configuration next to osd configuration only on debug mode

### PR DESCRIPTION
**Pull request by lightningterror**
Still needs a bit of work, will probably match the names with Linux gui.

image

Will leave this as a reference for future use so I have something to look at.
image

### CODE REVIEW

**Comment by lightningterror** [comment]
I'm probably forgetting something, the spin buttons work in reverse and go in to a negative number.

**Comment by turtleli** [comment]
So why are there 6 spin controls?

As for the spin controls not working properly, that's probably because the value you put in is too large to fit in a 16-bit integer. The documentation at https://docs.microsoft.com/en-us/windows/desktop/controls/bumper-up-down-control-reference-messages should point you to the solution for that.

**Comment by lightningterror** [comment]
Cool , thanks. Didn't know that was the issue.

I didn't have a ss when I started making this, will probably drop those extra spin buttons.

**Comment by lightningterror** [comment]
Updated. Should be better now.

image

**Comment by lightningterror** [comment]
Ups, messed up the ss, anyway updated.

Re upload 
image

**Comment by turtleli** [comment]
I think it'd be better if you use the same layout that is used on Linux.

I also think it'd be better if the spin control ranges are changed so they match what Linux uses.

**Comment by lightningterror** [comment]
image

> I also think it'd be better if the spin control ranges are changed so they match what Linux uses.

How to do that ?

**Comment by turtleli** [comment]
> How to do that ?

Did you look at the documentation link I provided above? I think it'd be better for your coding if you try and work out how to do it from that instead of having me tell you how to do it.

File: plugins/GSdx/GSSettingsDlg.cpp [hunk 1]
```diff
@@ -1015,5 +1030,81 @@ bool GSOSDDlg::OnMessage(UINT message, WPARAM wParam, LPARAM lParam)
...
+	{
+	case WM_COMMAND:
+	{
+		int id = LOWORD(wParam);
+
+		switch (id)
+		{
+		case IDOK:
+		{
+			INT_PTR data;
```
Comment (turtleli) [inline plugins/GSdx/GSSettingsDlg.cpp]: This is unused.

File: plugins/GSdx/GSSettingsDlg.cpp [hunk 2]
```diff
@@ -1015,5 +1030,81 @@ bool GSOSDDlg::OnMessage(UINT message, WPARAM wParam, LPARAM lParam)
...
+			theApp.SetConfig("debug_opengl", (int)IsDlgButtonChecked(m_hWnd, IDC_GL_DEBUGLOG));
+			theApp.SetConfig("debug_glsl_shader", (int)IsDlgButtonChecked(m_hWnd, IDC_GL_DEBUGGLSL));
+			theApp.SetConfig("dump", (int)IsDlgButtonChecked(m_hWnd, IDC_DUMP_GS));
+			theApp.SetConfig("save", (int)IsDlgButtonChecked(m_hWnd, IDC_DUMP_SAVE));
+			theApp.SetConfig("savef", (int)IsDlgButtonChecked(m_hWnd, IDC_DUMP_SAVEF));
```
Comment (turtleli) [inline plugins/GSdx/GSSettingsDlg.cpp]: Are the casts necessary?

File: plugins/GSdx/GSSettingsDlg.h [hunk 3]
```diff
@@ -40,6 +40,18 @@ class GSShaderDlg : public GSDialog
 	GSShaderDlg();
 };
 
+class GSDebugDlg : public GSDialog
+{
+	void UpdateControls();
```
Comment (turtleli) [inline plugins/GSdx/GSSettingsDlg.h]: I don't think this function is necessary if it does nothing.

File: plugins/GSdx/GSSettingsDlg.cpp [hunk 4]
```diff
@@ -1015,5 +1030,81 @@ bool GSOSDDlg::OnMessage(UINT message, WPARAM wParam, LPARAM lParam)
...
+			theApp.SetConfig("savet", (int)IsDlgButtonChecked(m_hWnd, IDC_DUMP_SAVET));
+			theApp.SetConfig("savez", (int)IsDlgButtonChecked(m_hWnd, IDC_DUMP_SAVEZ));
+			theApp.SetConfig("savel", (int)(SendMessage(GetDlgItem(m_hWnd, IDC_DUMP_SAVEL), UDM_GETPOS, 0, 0)));
+			theApp.SetConfig("saven", (int)(SendMessage(GetDlgItem(m_hWnd, IDC_DUMP_SAVEN), UDM_GETPOS, 0, 0)));
+
+			EndDialog(m_hWnd, id);
+		}
```
Comment (turtleli) [inline plugins/GSdx/GSSettingsDlg.cpp]: Missing break.

File: plugins/GSdx/GSSettingsDlg.cpp (hunk 2 above)
Comment (lightningterror) [reply plugins/GSdx/GSSettingsDlg.cpp]: I followed the other windows as an example.

Comment (ssakash) [reply plugins/GSdx/GSSettingsDlg.cpp]: I'd prefer if redundant casts are removed.

**Comment by lightningterror** [comment]
I'll finish this up another time, I spotted some other issues as well.

**Pull request closed without merging**